facebookresearch/ParlAI
# Feature: Add the ability to log model artifacts to Weights & Biases

**Patch description**

This PR adds the ability to log model artifacts to [Weights & Biases](https://wandb.ai/site). Logging model artifacts allows complete reproducibility of the experiments that resulted in the trained model and also allows to reuse the model for finetuning, evaluation and collaboration.

**Testing steps**
Run the demo training example with the optional arguments to log to wandb.

```
parlai train_model -t personachat \
-m transformer/ranker \
-mf /tmp/model_tr6 \
--n-layers 1 \
--embedding-size 300 \
--ffn-size 600 \
--n-heads 4 \
--num-epochs 2 \
-veps 0.25 \
-bs 64 \
-lr 0.001 \
--dropout 0.1 \
--embedding-type fasttext_cc \
--candidates batch \
--wandb-log True \
--wandb-project parlai \
--wandb-log-model True
```

This should log the model, dictionary, and the trainstats file to Weights & Biases. [Here](https://wandb.ai/parambharat/parlai/artifacts/model/smooth-capybara-11-model_tr8/v6/usage) is an example of the logged artifacts from the above command.
